src/history/api.py

```diff
@@ -397,3 +398,45 @@ def pdf_exists(year: str | int,
         bool: True if and only if the PDF file exists.
     """
     return Path.is_file(get_pdf_filepath(year, semester, student_type, round_num))
+
+
+def get_latest_year_and_sem_with_coursereg_info() -> tuple[str, str]:
```

Comment:
The function name is a bit long, do you think `get_latest_year_and_sem_with_data` is any better?